src/core/coap/coap_message.cpp

```diff
@@ -187,6 +187,23 @@ otError Message::AppendUriPathOptions(const char *aUriPath)
     return error;
 }
 
+otError Message::AppendBlockOption(uint8_t aType, uint32_t aNum, bool aMore, otCoapBlockSize aSize)
+{
+    otError  error   = OT_ERROR_NONE;
+    uint32_t encoded = aSize;
+
+    VerifyOrExit(aNum < 0xFFFFF, error = OT_ERROR_INVALID_ARGS);
+
+    encoded |= static_cast<uint32_t>((aMore) << 3);
+    encoded |= (aNum) << 4;
+
+    VerifyOrExit(aType == 1 || aType == 2, error = OT_ERROR_INVALID_ARGS);
```

Comment:
I wonder if an `enum` might be a clearer way of ensuring this… maybe in `coap_message.hpp` (since this isn't exposed to the C API):

```
typedef enum otCoapBlockType {
    OT_COAP_BLOCK_TYPE_1 = 1,
    OT_COAP_BLOCK_TYPE_2 = 2
} otCoapBlockType;
```

Then the compiler will enforce type 1 or type 2 for you.

Reply:
Fair, that's a good change. Although only a C++ compiler will enforce that. A C compiler doesn't differentiate an integer or an enum. In practice a user could provide any integer and not get an error.